[ROCm] Fixing Unit Test TestExportOnFakeCuda.test_fake_export

rahjain-amd opened the pull request:
Followup on https://github.com/pytorch/pytorch/pull/169941
forget to remove skipIfRocm decorator

Fixed the failures exposed by CI and enabled some Unexpected success cases


Tested with:
```sh
 PYTORCH_TEST_WITH_ROCM=1 pytest test/export/test_export_opinfo.py
```

<img width="1914" height="426" alt="image" src="https://github.com/user-attachments/assets/10b5c852-3473-44bb-90b1-70ad01223306" />


cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @jataylo @hongxiayang @naromero77amd @pragupta @jerrymannil @xinyazhang

rahjain-amd committed:
[ROCm] Fixing Unit Test TestExportOnFakeCuda.test_fake_export

Followup on #169941

forget to remove skipIfRocm decorator

jeffdaily reviewed (approved):
approved, pending CI

jeffdaily commented:
@pytorchbot rebase

rahjain-amd committed:
Fixing - TestExportOnFakeCudaCUDA.test_fake_export_nn_functional_scaled_dot_product_attention_cuda_float32

CI exposed this issue.

This also fix below cases which are expected to fail

```py
    # xfail("nn.functional.grid_sample"),
    # xfail("to_sparse"),
    # following are failing due to OptionalDeviceGuard
    # xfail("__getitem__"),
    # xfail("nn.functional.batch_norm"),
    # xfail("nn.functional.instance_norm"),
    # xfail("nn.functional.multi_margin_loss"),
    # xfail("nonzero"),
```

rahjain-amd committed:
Readd the xfail tests

rahjain-amd committed:
Merge branch 'pytorch:main' into fix-ut-168551-2
